server/src-rsr/migrations/37_to_38.sql

```diff
@@ -0,0 +1,49 @@
+CREATE FUNCTION hdb_catalog.current_setting(setting_name text) RETURNS json
+  LANGUAGE plpgsql AS $$
+  DECLARE
+    server_version_num int;
+    setting_value text;
+  BEGIN
+    server_version_num := current_setting('server_version_num');
+    IF server_version_num >= 90600 THEN
+      setting_value := current_setting('hasura.' || setting_name, true);
+    ELSE
+      -- We still support Postgres 9.5, which doesn’t support the second
+      -- argument to current_setting, so emulate it by just catching the error.
+      BEGIN
+        setting_value := current_setting('hasura.' || setting_name);
+      EXCEPTION WHEN OTHERS THEN
+        setting_value := NULL;
+      END;
+    END IF;
+    -- Note: we test for the empty string here to work around a Postgres bug.
+    -- When SET LOCAL is used to set the value of a setting that does not
+    -- currently exist, Postgres implicitly creates a /session-wide/ definition
+    -- of the setting with the empty string as its value. This is annoying, but
+    -- fortunately an empty string is not a legal JSON value, so it always means
+    -- no value was set in the current transaction.
+    IF setting_value IS NULL OR setting_value = '' THEN
+      RETURN NULL;
+    ELSE
+      RETURN setting_value::json;
+    END IF;
+  END $$;
+
+CREATE OR REPLACE FUNCTION hdb_catalog.insert_event_log(schema_name text, table_name text, trigger_name text, op text, row_data json)
+  RETURNS text LANGUAGE plpgsql AS $$
+  DECLARE
+    id text;
+    payload json;
+  BEGIN
+    id := gen_random_uuid();
+    payload := json_build_object(
+      'op', op,
+      'data', row_data,
+      'session_variables', hdb_catalog.current_setting('user'),
+      'trace_context', hdb_catalog.current_setting('tracecontext')
+    );
+    INSERT INTO hdb_catalog.event_log
+                (id, schema_name, table_name, trigger_name, payload)
+      VALUES (id, schema_name, table_name, trigger_name, payload);
+    RETURN id;
+  END $$;
```

Comment:
Do we need this? Because there doesn't seem to be an equivalent change in master to this.

Reply:
See the discussion [here](https://hasurahq.slack.com/archives/CV3UR1MT2/p1596570592148400) resulting in commit 66e85ab9f.

Reply:
This is a workaround for a postgres bug with tracing (see the Note). It is very recent. I thought there was a plan to push it to master too, but maybe others agreed for it to be merged alongside PDV?

Reply:
I think this issue is solved in master via [this commit](https://github.com/hasura/graphql-engine/commit/b2561a719bb586a74dc40515a6f4637c79b91f03). (Although, I'm not a 100% sure)

@paf31 Can you confirm?

Reply:
That commit you mention seems unrelated to me. And yeah, master doesn't have the new metadata version (I had to cherry-pick this commit to be able to run my local DB with both branches). So we do really want this patch. ^^
(will PM you the link to the Slack thread where this was discussed)

Reply:
The fix was to make sure `hasura.tracecontext` was defined on all code paths, including inserts via `v1/query`. Maybe that's why it doesn't look related. There was no update to the function.